### PR TITLE
implementation of Gaussian Elimination pivoting as a numerical linear algebra algorithm

### PR DESCRIPTION
**Pull request by TheKidPadra**
### Describe your change:

This is my first ever contribution to Hacktoberfest 2023. I hope I did everything right.
I'm kind of a newbie to Hacktoberfest.
And If I did anything wrong, please tell me, and I'll fix it right away.
I would really appreciate that
Thank you for this opportunity

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: linear_algebra/src/gaussianeliminationpivoting.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import numpy as np
+from typing import List
+
+def custom_pivoting(a: np.ndarray, n: int, i: int) -> int:
```
[the remark this answers is not in the document]
Comment (TheKidPadra) [reply linear_algebra/src/gaussianeliminationpivoting.py:5]: I already provided a descriptive name for every parameter

**Comment by TheKidPadra** [comment]
Can someone help me I did everything and it says to use list instead of List and it's not possible 
the second thing is I already provided a descriptive name for every parameter but it still says no

**Comment by cclauss** [comment]
Click the `Details` links to the right of the failing tests ❌ below for more info.

**Comment by TheKidPadra** [comment]
> Click the `Details` links to the right of the failing tests ❌ below for more info.

Thank you, dear [cclauss](https://github.com/cclauss),
Finally, all checks have passed
Waiting for a Review

**Comment by TheKidPadra** [comment]
Only 5 days left to end of October can someone please review this code?

**Comment by cclauss** [comment]
> Please provide a descriptive name for the parameter: `a`

As discussed in `CONTRIBUTING.md` we want to avoid single-letter variable names.  Please use self-documenting variable names to help your reader understand the algorithm.

**Comment by TheKidPadra** [comment]
I changed 'a' to 'matrix' and 'coeff_matrix' for better clarity

File: linear_algebra/src/gaussian_elimination_pivoting.py [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import numpy as np
 
-
```
Comment (cclauss) [inline linear_algebra/src/gaussian_elimination_pivoting.py]: Please put this blank line back.

Comment (TheKidPadra) [reply linear_algebra/src/gaussian_elimination_pivoting.py]: yes sir

**Comment by TheKidPadra** [comment]
All things are fixed, please review it before the end of Hacktoberfest.
I changed 'a' to matrix and 
I changed 'n' to num_rows
I changed 'i' to column_index
for better clarity

**Comment by TheKidPadra** [comment]
It says FileNotFoundError: matrix.txt not found.

**Comment by TheKidPadra** [comment]
It says FileNotFoundError: matrix.txt not found 
but matrix.txt is in the same folder as the source code 
can you help me?

File: linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import sys
+import time
```
Comment (cclauss) [inline linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py:2]:
```suggestion
import time
from pathlib import Path
```

File: linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+import numpy as np
+
+matrixab = np.loadtxt("matrix.txt")
```
Comment (cclauss) [inline linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py:6]:
```suggestion
matrixab = np.loadtxt(Path(__file__).parent / "matrix.txt")
```

File: linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+B = np.copy(matrixab[:, matrixab.shape[1] - 1])
+
+
+def foo(matrix):
```
Comment (cclauss) [inline linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py:10]: Please.  Come up with a better name for this function and add a type hint and maybe even some doctests.

According to `CONTRIBUTING.md` algorithmic functions should not `print()`.

File: linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    print("x vector: ")
+    print(xlst)
+    print(" ")
+    print(f"Start time: {start}, End time: {stop}")
+    print(f"Elapsed time during the whole function in seconds: {stop - start}")
+
+    return np.asarray(xlst)
+
+
+vectorofxalpha = foo(matrixab)
```
Comment (cclauss) [inline linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py:65]:
```suggestion
if __name__ == "__main__":
    vectorofxalpha = foo(matrixab)
```
And indent all lines below in this file.

File: linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py (hunk 5 above)
Comment (TheKidPadra) [reply linear_algebra/src/gaussian_elimination_pivoting/gaussian_elimination_pivoting.py]: Function is renamed to solve_linear_system, and type hints are added for parameters and return values. The doctest provides an example of usage and asserts the correctness of the solution. I've also removed the print statements and replaced the exit call with a sys.exit exception to allow for better error handling.

**Comment by TheKidPadra** [comment]
I did everything but it still doesn't pass build test 
but It's working in VScode

**Comment by TheKidPadra** [comment]
Will you accept it?

**Comment by TheKidPadra** [comment]
Your right sir. I don't know what was I thinking.
Here, finally it got fixed

**Review by cclauss — APPROVED** [review]

**Pull request merged**